### PR TITLE
Return bad request if no token provided

### LINKED ISSUE #135: Solve TypeError in token check
It's not a bug in behavior really but filling our logs unnecessary.

### PR DESCRIPTION
**Pull request by nmlinaric**
Return 400 - bad request if no token is provided.

Resolves #135 Solve TypeError in token check.

### CODE REVIEW

File: src/Middleware/Authorization.ts [hunk 1]
```diff
@@ -13,8 +13,10 @@ export async function AuthorizeUser(
 ) {
     try {
         let token = (req.headers.authorization || req.headers.Authorization) as string;
-        if (token.startsWith('Bearer ')) {
+        if (token?.startsWith('Bearer ')) {
             token = token.slice(7, token.length).trimLeft();
+        } else {
+            throw new ServiceError(400, 'Bad request.')
```
Comment (morrigan) [inline src/Middleware/Authorization.ts:19]: Not really a good solution as you just proposed that token MUST start with `Bearer `. We have been loose on that standard, on frontend we are not even sending that I think.

I think the easy solution would be to check if token exists before this condition.

Comment (nmlinaric) [reply src/Middleware/Authorization.ts:19]: Should I then omit the `Bearer token` check and just check `if(!token)`?

Comment (morrigan) [reply src/Middleware/Authorization.ts:19]: This is not the check only. This cleans the string from`Bearer ` if it's sent and only uses token later

**Comment by morrigan** [comment]
@nmlinaric Maybe would be good to remove `log.error` from that middleware. Or at least to be info or warn

**Pull request merged**